components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/util/IdPSecretsProcessor.java

```diff
@@ -16,32 +16,32 @@
  * under the License.
  */
 
-package org.wso2.carbon.identity.secret.mgt.core;
+package org.wso2.carbon.idp.mgt.util;
 
 import com.google.gson.Gson;
 import org.apache.commons.lang.StringUtils;
 import org.wso2.carbon.identity.application.common.model.FederatedAuthenticatorConfig;
 import org.wso2.carbon.identity.application.common.model.IdentityProvider;
 import org.wso2.carbon.identity.application.common.model.Property;
+import org.wso2.carbon.identity.secret.mgt.core.SecretManager;
+import org.wso2.carbon.identity.secret.mgt.core.SecretResolveManager;
+import org.wso2.carbon.identity.secret.mgt.core.SecretsProcessor;
 import org.wso2.carbon.identity.secret.mgt.core.exception.SecretManagementException;
 import org.wso2.carbon.identity.secret.mgt.core.model.ResolvedSecret;
 import org.wso2.carbon.identity.secret.mgt.core.model.Secret;
 import org.wso2.carbon.identity.secret.mgt.core.model.SecretType;
+import org.wso2.carbon.idp.mgt.internal.IdpMgtServiceComponentHolder;
+
 import static org.wso2.carbon.identity.secret.mgt.core.constant.SecretConstants.IDN_SECRET_TYPE_IDP_SECRETS;
 
 /**
  * Identity provider secrets processor service implementation.
  */
 public class IdPSecretsProcessor implements SecretsProcessor<IdentityProvider> {
 
-    private final SecretManager secretManager;
-    private final SecretResolveManager secretResolveManager;
     private final Gson gson;
 
     public IdPSecretsProcessor() {
-
-        this.secretManager = new SecretManagerImpl();
-        this.secretResolveManager = new SecretResolveManagerImpl();
```

Comment:
IMO, it's better to keep these references and refer the singleton instances here as each is used in multiple places over the method calls. IMO it reduce unnecessary method calls as well as improves the readability and maintainability

Reply:
When creating the instance of the IdPSecretsProcessor class, there can be a possibility to not to set the SecertManager and the SecretResolveManager to the `IdpMgtServiceComponentHolder` from the OSGI services.

Reply:
Addressed with commit b94ac1b72866b61aa03c580e3837007c501fce29